pyvista/plotting/render_window_interactor.py

```diff
@@ -433,26 +437,31 @@ def key_press_event(self, *args):
                 func()
 
     def update_style(self):
-        """Update the camera interactor style."""
-        if self._style_class is None:
-            # We need an actually custom style to handle button up events
-            self._style_class = _style_factory(self._style)(self)
+        """Update the camera interactor style.
+
+        Called when setting interactor_style attribute.
+        """
         self.interactor.SetInteractorStyle(self._style_class)
 
     @property
-    def style(self):
+    def style(self) -> Optional[Union[_vtk.vtkContextInteractorStyle, _vtk.vtkInteractorStyle]]:
         """Return the current interactor style.
 
         Returns
         -------
-        vtkInteractorStyle
+        vtkInteractorStyle or vtkContextInteractorStyle or None
             The current interactor style.
 
         """
-        if self._style_class is None:
-            self.update_style()
         return self._style_class
 
+    @style.setter
+    def style(self, style: _vtk.vtkInteractorStyle):
+        # TODO If a plain vtk class, wrap it in pyvista class???
```

Comment:
Always wrapping with custom pyvista functionality might lower flexibility for users?  Additionally, context style interactor functionality currently uses the private `_style_class` attribute, but if this changes in the future to use this setter, we would need to not modify the style passed in here.